torch_geometric/datasets/infection_dataset.py

```diff
@@ -0,0 +1,163 @@
+        assert len(num_infected) == num_graphs
+        assert all(num_infected) >= 0
+
+        self.graph_generator = GraphGenerator.resolve(
+            graph_generator,
+            **(graph_generator_kwargs or {}),
+        )
+        self.seeds = seeds
+        self.max_path_length = max_path_length
+        self.num_infected = num_infected
+
+        data_list = [
+            self.get_graph(
+                path_length=self.max_path_length[i],
+                num_infected=self.num_infected[i],
+                seed=self.seeds[i] if self.seeds is not None else None)
+            for i in range(num_graphs)
+        ]
+        self.data, self.slices = self.collate(data_list)
+
+    def get_graph(self, path_length: int, num_infected: int,
+                  seed: Optional[int] = None) -> Explanation:
+        import networkx as nx
+        if seed is not None:
+            seed_everything(seed)
+        data = self.graph_generator()
+        num_nodes, edge_index = data.num_nodes, data.edge_index
+        rand_perm = torch.randperm(num_nodes)
+
+        x = torch.zeros((num_nodes, 2), dtype=torch.float32)
+        x[rand_perm[:num_infected], 1] = 1  # Infected
+        x[rand_perm[num_infected:], 0] = 1  # Healthy
+        y = torch.full((num_nodes, ), path_length, dtype=torch.long)
+        node_mask = torch.zeros(num_nodes, dtype=torch.long)
+        explain_path_edge_index = []
+        # Original infected nodes with label 0
+        y[x[:, 1].nonzero(as_tuple=True)[0]] = 0
+        G = to_networkx(Data(edge_index=edge_index))
+        for node in range(num_nodes):
+            for distance in range(1, path_length):
+                if x[node, 1] == 1 or G.in_degree(node) <= 0:
+                    break
+                # Get the subgraph with nodes that reach to node with k hops
+                sub_nodes, _, _, _ = k_hop_subgraph(node, distance, edge_index)
+                # All nodes in the subgraph are healthy
+                if sum(x[sub_nodes, 1]) == 0:
+                    continue
+                infected_indices = x[sub_nodes, 1].nonzero(as_tuple=True)[0]
+                infected_nodes = sub_nodes[infected_indices]
+                sp_length = [(idx.item(),
+                              nx.shortest_path_length(G, idx.item(), node))
+                             for idx in infected_nodes]
+                # only retain the shortest path with distance length
+                sp_length = [(idx, length) for idx, length in sp_length
+                             if length == distance]
+                if len(sp_length) == 0:
+                    continue
+                elif len(sp_length) == 1:
+                    # Add explanation for node with one unique path
+                    src_node = sp_length[0][0]
+                    sp_generator = nx.all_shortest_paths(G, src_node, node)
+                    sp = next(sp_generator)
+                    if next(sp_generator, None) is None:
```

Comment:
for nodes with multiple paths, is there no groundtruth explanation?

Reply:
Yes, that's what the paper said